src/lib/Elab.ml

```diff
@@ -43,19 +43,19 @@ struct
     | Ref (key, sb') -> Ref (key, intoS @@ Cmp (sb, sb'))
 end
 
-module Elab : Elab 
+module Elab : Elab
   with type term = Tm.term
   and type subst = Tm.subst
   and type hole = Tm.hole
-= 
+=
 struct
   type hole = Tm.hole
   type term = Types.term
   type subst = Types.subst
 
-  module HT = Caml.Hashtbl.Make (String)
+  type env = (string, term jdg) Hashtbl.t
```

Comment:
much better!

Btw, how does the polymorphic version know how to compute hash keys from an arbitrary type? Is this some ocaml magic or something?

Reply:
Ahh, it uses first class modules when initializing? cool.

Reply:
So for this version you will still need to pack up a module that supports the comparison and hashing stuff when you create the table. Most of the data specific modules in `base` already provide this, like `String` and `Int`:

```ocaml
let table = Hashtbl.create (module String) ()
```

There is also the `Hashtbl.Poly` version [here](https://ocaml.janestreet.com/ocaml-core/latest/doc/base/Base/Hashtbl/Poly/) which is maybe what you mean. That one seemingly reuses the same magic that the `Caml` lib uses for polymorphic comparison and hashing. I'm not entirely sure how that stuff works though and I usually try to avoid it.